libs/langchain-google-genai/src/tests/chat_models-extended.int.test.ts

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable no-process-env */
+import { test, expect } from "@jest/globals";
+import { z } from "zod";
+import { ChatGoogleGenerativeAI } from "../chat_models.js";
+
+const { GOOGLE_API_KEY } = process.env;
+if (!GOOGLE_API_KEY) {
+  throw new Error("Cannot run tests because GOOGLE_API_KEY is not set.");
+}
+
+const baseSchema = z.object({
+  name: z.string(),
+  age: z.number(),
+});
+
+test("Google AI - Generate structured output without errors", async () => {
+  const model = new ChatGoogleGenerativeAI({
+    model: "gemini-1.5-flash",
+    temperature: 0.7,
+  });
+  const structuredLlm = model.withStructuredOutput(baseSchema);
+  const request = "Generate a structured response for a user.";
+  const result = await structuredLlm.invoke(request);
+  console.log("Valid Schema Result:", result);
+  expect(result).toBeDefined();
+  expect(result).toHaveProperty("name");
+  expect(result).toHaveProperty("age");
+});
+
+test("Google AI - Throw error if output does not match schema", async () => {
+  const schema = z.object({ name: z.string(), age: z.string() }); // Schema mismatch
+  const model = new ChatGoogleGenerativeAI({
+    model: "gemini-1.5-flash",
+    temperature: 0.7,
+  });
+  const structuredLlm = model.withStructuredOutput(schema);
+  const request = "Generate structured data where age is a number.";
```

Comment:
This doesn't seem like a reliable test - shouldn't we expect an error to be thrown? Going to delete this one.